### DISCUSSION TITLE
Group administration and privacy

### DISCUSSION THREAD

**Discussion by hughrun** [bookwyrm-social/bookwyrm#1549] [Ideas]
With #1490 users can now join a Group. For now all that allows them to do is share Lists, but additional features can be added in future.

1. It seems like it might be useful to provide an option for Group administrators with authorisation to do special things acting on behalf of the group, such as:

- invite and remove members
- delete the group (maybe)
- make announcements
- make polls and events

2. Related to this, it could be useful to allow ownership of the group (the `group.user`) to be transferred to another user. For example a group owner may wish to delete their account (either because they are moving servers/identities, or because they no longer wish to use Bookwyrm), but doesn't want the group to be wiped out because other people are enjoying it.

   Transferring ownership could require some thinking about how we deal with privacy settings, however (see below).

3. Group privacy probably needs more consideration. Think about the following scenario:

   * Alex is the owner of the Forest Lovers group. The Forest Lovers group has `Followers only` visibility. 
   * Alex requires follow requests and has a small, carefully managed list of followers. Alex transfers ownership of the Forest Lovers group to Sam.
   * Sam allows anyone to follow them, and has a large list of followers.

   For a third user, Ashley, this potentially creates a dilemma. Ashley trusts Alex's judgement and is comfortable with Alex's followers seeing that they love forests - but is Ashley equally happy for all of Sam's followers to see that they are a member of this group? We don't know.

   The set up of group privacy was intended to follow the already-existing model used for Lists in order to keep things simple and familiar to users, but perhaps this was a mistake. Questions to consider:

   1. Do we need any group visibility/privacy options other than `public` or `private`? i.e.  are `unlisted` and `followers` actually meaningful or useful privacy levels for groups?
   2. If we keep `followers`, should this work in the context of the user being viewed? That is, if I am looking at a User page then I only see any groups they're a member of if they are the owner, so the current model works fine, but if I am viewing the Group page, what should determine whether I can see that a user is a member of the group? Is it merely my relationship to the group owner (determining whether or not I can see the page at all), or to each individual group member (determining whether I can see that a particular user is a member of the group)? Currently only the relationship with the group owner is considered.
   3. Related to (2), should user blocks have any bearing on visibility of a group's membership? e.g. let's say a fourth user, Lesley, is blocked by Ashley. Should Lesley be able to see that Ashley is a member of the group when viewing the group's homepage? I can see both that it would be useful to "hide" group membership for privacy reasons, but also that it could be problematic if the blockee assumes that means they're welcome to join the group because the blocker isn't a member. Currently all users are visible but if you try to visit the user page of someone who blocked you, that's not possible.

**Comment by mouse-reeve** [comment]
Another consideration - it may make sense for there to be multiple people who are admins of a group, which would makes "followers" even more complex.

**Reply by hughrun** [reply]
Privacy implications resolved in #1561 but the concept of admins remains as a future feature to implement.
